File: exercise-06-retirement-calculator/retirement_calculator.py
from datetime import datetime

current_age = int(input("What is your current age? "))
retire_age = int(input("At what age would you like to retire? "))

years_left = retire_age - current_age
current_year = datetime.now().year
retirement_year = current_year + years_left

if years_left < 1:
  print("You can already retire.")
else:
  print(f'You have {years_left} years left until you can retire.\nIt\'s {current_year}, so you can retire in {retirement_year}.')

File: exercise-06-retirement-calculator/RetirementCalculator.java
import java.time.Year;

class RetirementCalculator {
  public static void main(String[] args) {
    var line = System.console().readLine("What is your current age? ");
    var currentAge = Integer.parseInt(line);

    line = System.console().readLine("At what age would you like to retire? ");
    var retireAge = Integer.parseInt(line);

    var yearsLeft = retireAge - currentAge;
    var currentYear = Year.now().getValue();
    var retirementYear = currentYear + yearsLeft;

    if (yearsLeft < 1) {
      System.out.println("You can already retire.");
    } else {
      System.out.printf("You have %d years left until you can retire.%nIt's %d, so you can retire in %d.%n",
        yearsLeft, currentYear, retirementYear);
    }
  }
}
